pkg/devfile/adapters/common/utils.go

```diff
@@ -94,3 +104,43 @@ func GetVolumes(devfileObj devfileParser.DevfileObj) map[string][]DevfileVolume
 	}
 	return componentAliasToVolumes
 }
+
+// IsEnvPresent checks if the env variable is present in an array of env variables
+func IsEnvPresent(envVars []common.DockerimageEnv, envVarName string) bool {
+	for _, envVar := range envVars {
+		if *envVar.Name == envVarName {
+			return true
+		}
+	}
+
+	return false
+}
+
+// IsPortPresent checks if the port is present in the endpoints array
+func IsPortPresent(endpoints []common.DockerimageEndpoint, port int) bool {
+	for _, endpoint := range endpoints {
+		if *endpoint.Port == int32(port) {
+			return true
+		}
+	}
+
+	return false
+}
+
+// IsComponentBuildRequired checks if a component build is required based on the push commands, it throws an error
+// if the push commands does not meet the expected criteria
+func IsComponentBuildRequired(pushDevfileCommands []common.DevfileCommand) (bool, error) {
+	var buildRequired bool
+
+	if len(pushDevfileCommands) == 1 {
+		// if there is one command, it is the mandatory run command. No need to build.
+		buildRequired = false
+	} else if len(pushDevfileCommands) == 2 {
+		// if there are two commands, it is the optional build command and the mandatory run command, set buildRequired to true
+		buildRequired = true
+	} else {
```

Comment:
This entire section you can use a `case` instead of if statements

Ex:
```go
switch len(pushDevfileCommands) {
default:
	...
case 1:
	...
case 2:
	...
}
```